src/env/mod.rs

```diff
@@ -2,14 +2,35 @@ use crate::api::customization::Customization;
 use crate::api::firmware_protection::FirmwareProtection;
 use crate::api::upgrade_storage::UpgradeStorage;
 use crate::ctap::status_code::Ctap2StatusCode;
-use crate::ctap::Channel;
+use crate::ctap::{Channel, Transport};
 use persistent_store::{Storage, Store};
 use rng256::Rng256;
 
+#[cfg(feature = "std")]
+pub mod host;
 #[cfg(feature = "std")]
 pub mod test;
 pub mod tock;
 
+pub enum SendOrRecvStatus {
+    // FIXME: Replace Option<SendOrRecvStatus>::None with SendOrRecvStatus::TimedOut?
+    Error,
+    Sent,
+    Received(Transport),
+}
+
+pub trait IOChannel {
+    // FIXME: Separate type for timeout.
+    fn recv_with_timeout(&mut self, buf: &mut [u8; 64], timeout: isize)
+        -> Option<SendOrRecvStatus>;
```

Comment:
We can remove receive for now, since it's not needed for `check_user_presence`. Note that by removing this function, we open ourselves to having 2 channels and not passing a `Transport`.

HID receive is special in a sense that you can NOT choose the channel you are receiving on. Maybe we will eventually have a function that receives on any channel and we can then multiplex the packets, but we don't need to make that decision yet.